R/set-record-option.R

```diff
@@ -0,0 +1,110 @@
+
+#' Add or modify a record option
+#'
+#' Add a new record option or adjust the value/presence of an existing one.
+#'
+#' Note that only estimation, data, and table records are currently supported.
+#'
+#' Notes on `value`:
+#'  - We infer whether to use `option_value` or `option_flag` for *new* options based on the *class* of `value`:
+#'     - If you are trying to add an option with an assigned value (i.e. `MAXEVAL = 10`), `value` must be a character string or numeric value
+#'     - If you are trying to add a flag option (i.e. `NOABORT`), `value` should be logical (`T`/`F`).
```

Comment:
* Please spell out `TRUE` and `FALSE` rather than recommend anyone use `base::T` or `base::F` (which may be masked).

* `NULL` is the only thing documented in `?option` as "don't render".  Please stick with that.  Whether we should change `FALSE` to mean that too can be a broader discussion, but I'd like this PR should stay consistent with the current behavior of `opt$value <- ...` handling (i.e. please remove the `FALSE` special treatment).